mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/rules/Moves.kt

```diff
@@ -142,3 +147,91 @@ fun Board<Piece<Role>>.diagonals(
     )
   }
 }
+
+/** Moves representing a castling to the left. */
+fun BoardWithHistory<Piece<Role>>.leftCastling() =
+    castling(
+        kingStart = Position(x = 4, y = 7),
+        kingEnd = Position(x = 2, y = 7),
+        rookStart = Position(x = 0, y = 7),
+        rookEnd = Position(x = 3, y = 7),
+        empty = setOf(Position(x = 1, y = 7), Position(x = 2, y = 7), Position(x = 3, y = 7)),
+    )
+
+/** Moves representing a castling to the right. */
+fun BoardWithHistory<Piece<Role>>.rightCastling() =
+    castling(
+        kingStart = Position(x = 4, y = 7),
+        kingEnd = Position(x = 6, y = 7),
+        rookStart = Position(x = 7, y = 7),
+        rookEnd = Position(x = 5, y = 7),
+        empty = setOf(Position(x = 5, y = 7), Position(x = 6, y = 7)),
+    )
```

Comment:
Either names are fine imho. I'd suggest we keep the `left` and `right` names for the moment (especially since this is an "implementation detail" of how the `Game` checks for valid piece movements, and it's never used outside of the model), but we may revisit that decision later.